### PR TITLE
fix: enforce user must specify access_key and secret_key using aws auth

### LINKED ISSUE #11086: bug: credentials for external sources should be enforced
### Describe the bug

For now, the credentials can be omitted, e.g,.

```sql
CREATE TABLE side_input(
      key BIGINT PRIMARY KEY,
      value VARCHAR
    )  WITH (
    connector = 's3',
    s3.region_name = 'us-east-1',
    s3.bucket_name = 'xxxx',
    match_pattern = 'side-input.csv'
    ) FORMAT PLAIN ENCODE CSV (
        without_header = 'true',
        delimiter = ','
    );
```

, and RisingWave will use the local AWS credentials to access the S3 bucket.

It might be handy for local deployments but it's dangerous when RisingWave is serving external queries. There's a chance that the querier can have access to files in the RisingWave's state store.

### Error message/log

```text
ERROR:  QueryError: internal error: connector error: service error
```

### To Reproduce

_No response_

### Expected behavior

I expected to see that the credentials are enforced and the SQL is rejected because lacking of them, like what [Snowflake does](https://docs.snowflake.com/en/user-guide/data-load-s3-copy).

```sql
CREATE TABLE side_input(
      key BIGINT PRIMARY KEY,
      value VARCHAR
    )  WITH (
    connector = 's3',
    s3.region_name = 'us-east-1',
    s3.bucket_name = 'xxxx',
    s3.credentials.access='dummy',
    s3.credentials.secret='dummy',
    match_pattern = 'side-input.csv'
    ) FORMAT PLAIN ENCODE CSV (
        without_header = 'true',
        delimiter = ','
    );
```

### How did you deploy RisingWave?

_No response_

### The version of RisingWave

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

resolve #11086 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [x] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Connector (sources & sinks)

### Release note

access_key and corresponding secret_key become a must for all aws auth components. 

</details>

### CODE REVIEW

File: src/connector/src/aws_auth.rs
Comment (xxchan) [inline src/connector/src/aws_auth.rs:1]: Should we reject it in fe?

**Comment by huangjw806** [comment]
If the bucket is Publicly accessible, can we don't provide AK/SK?

File: src/connector/src/aws_auth.rs
Comment (xxchan) [reply src/connector/src/aws_auth.rs:1]: But might be ok as a quick fix

Comment (tabVersion) [reply src/connector/src/aws_auth.rs:1]: still discussing with @arkbriar, whether we should continuously support public buckets.

File: src/connector/src/aws_auth.rs [hunk 1]
```diff
@@ -127,7 +123,7 @@ impl AwsAuthProps {
     pub async fn build_config(&self) -> anyhow::Result<SdkConfig> {
         let region = self.build_region().await?;
         let credentials_provider = self
-            .with_role_provider(self.build_credential_provider().await?)
+            .with_role_provider(self.build_credential_provider()?)
             .await?;
         let config_loader = aws_config::from_env()
```
[the remark this answers is not in the document]
Comment (tabVersion) [reply src/connector/src/aws_auth.rs:128]: the problem is about finding an alternative for `from_env`.

> > Important: Using the aws-config crate to configure the SDK is preferred to invoking this builder directly. Using this builder directly won’t pull in any AWS recommended default configuration values.
> 
> ---- from [doc](https://docs.rs/aws-config/latest/aws_config/struct.SdkConfig.html#implementations)

It is not recommended to build `SdkConfig` directly.

File: src/connector/src/aws_auth.rs
Comment (fuyufjh) [reply src/connector/src/aws_auth.rs:1]: > still discussing with @arkbriar, whether we should continuously support public buckets.

Prefer to keep it. It's up to user's choice, not our fault.

**Review by fuyufjh — APPROVED** [review]

**Comment by tabVersion** [comment]
> If the bucket is Publicly accessible, can we don't provide AK/SK?

Just change to an empty string as default. We won't use our env var any more.

**Comment by tabVersion** [comment]
image

it looks like we cannot use a random or empty string as AK/SK, even for public access bucket.

**Review by arkbriar — APPROVED** [review]
LGTM!

**Pull request merged**